libs/utils/src/fs_ext.rs

```diff
@@ -75,7 +75,8 @@ mod test {
         );
 
         // invoke on a file to ensure it returns an error
-        let file_path: PathBuf = dir_path.join("testfile");
+        let file_path: Utf8PathBuf =
+            Utf8PathBuf::from_path_buf(dir_path.join("testfile")).expect("non-Unicode path");
```

Comment:
I wonder if it would make sense in this module to have a custom `tempdir()` which would return something which we know there is a Utf8PathBuf?

It would appear that the `tempfile::Builder` will always create an ascii paths.

Also, the author of camino has `camino_tempfile` which would already have the utilities. If I recall correclty, the tempfile==3.5.0 is what we depend on right now.

Reply:
Hmm, you mean that we should replace the `tempfile` crate with `camino_tempfile` crate.
And update the related codes? @koivunej

Reply:
Earlier me:

> I wonder if it would make sense in this module to have a custom tempdir() which would return something which we know there is a Utf8PathBuf?

Looking around, we do have plenty of usages for `tempfile::tempdir` in tests.

I was initially thinking of if this should be added into this tests module:

```patch
+    fn tempdir() -> Utf8Tempdir {
+        // FIXME: this could be solved with camino_tempfile dependency
+        let dir = tempfile::tempdir().unwrap();
+        Utf8Tempdir::new(dir)
+    }
+
+    struct Utf8Tempdir(tempfile::TempDir);
+
+    impl Utf8Tempdir {
+        fn new(dir: tempfile::TempDir) -> Self {
+            let ret = Self(dir);
+            ret.path();
+            ret
+        }
+
+        fn path(&self) -> &camino::Utf8Path {
+            use camino::Utf8Path;
+            match Utf8Path::from_path(self.0.path()) {
+                Some(p) => p,
+                None => {
+                    if Utf8Path::from_path(&std::env::temp_dir()).is_none() {
+                        panic!("std::env::temp_dir() is pointing to a non-utf8 directory so temporary directories on utf8 paths cannot be created")
+                    } else {
+                        unreachable!("tempfile::tempdir should create ascii paths but the std::env::temp_dir() is utf-8")
+                    }
+                }
+            }
+        }
+    }
```

But this is exactly what `camino_tempfile` provides, with adjusted docs.

I think it would make sense to use it. It would allow getting rid of repeated conversion code.